zebra/interface.c

```diff
@@ -1736,59 +1718,28 @@ DEFUN (show_interface_name_vrf,
 
 	interface_update_stats();
 
-	VRF_GET_ID(vrf_id, argv[idx_name]->arg, false);
-
-	/* Specified interface print. */
-	ifp = if_lookup_by_name(argv[idx_ifname]->arg, vrf_id);
-	if (ifp == NULL) {
-		vty_out(vty, "%% Can't find interface %s\n",
-			argv[idx_ifname]->arg);
-		return CMD_WARNING;
-	}
-	if_dump_vty(vty, ifp);
-
-	return CMD_SUCCESS;
-}
-
-/* Show specified interface to vty. */
-DEFUN (show_interface_name_vrf_all,
-       show_interface_name_vrf_all_cmd,
-       "show interface IFNAME [vrf all]",
-       SHOW_STR
-       "Interface status and configuration\n"
-       "Interface name\n"
-       VRF_ALL_CMD_HELP_STR)
-{
-	int idx_ifname = 2;
-	struct vrf *vrf;
-	struct interface *ifp;
-	int found = 0;
-
-	interface_update_stats();
+	if (argc > 3) {
+		VRF_GET_ID(vrf_id, argv[idx_name]->arg, false);
 
```

Comment:
as you unreferenced VRF_GET_ID in previous function, perhaps you can dereference it too ?
by using (see line 1677)

Reply:
if_lookup_by_name uses vrf_id, so I saved VRF_GET_ID usage.
I just pushed changes that introduce if_lookup_by_name_vrf which uses vrf pointer instead.